### PR TITLE
Fix #9500: Use `java.class.path` for Java 9+ class loaders

### LINKED ISSUE #9500: Java 9+: staging classpath handling
When run the test set with JDK 11, we get two errors:

(1) Failure in `project/scripts/bootstrapCmdTests`

```
+ ./project/scripts/sbt ';dotty-compiler-bootstrapped/dotc -with-compiler tests/run-staging/quote-run.scala; dotty-compiler-bootstrapped/dotr -with-compiler Test'
Exception in thread "main" dotty.tools.dotc.MissingCoreLibraryException: Could not find package scalaShadowing from compiler core libraries.
Make sure the compiler core libraries are on the classpath.

  at dotty.tools.dotc.core.Denotations$.select$1(Denotations.scala:1222)
  at dotty.tools.dotc.core.Denotations$.recurSimple$1(Denotations.scala:1250)
  at dotty.tools.dotc.core.Denotations$.recur$1(Denotations.scala:1252)
  at dotty.tools.dotc.core.Denotations$.staticRef(Denotations.scala:1254)
  at dotty.tools.dotc.core.Symbols$.requiredPackage(Symbols.scala:890)
  at dotty.tools.dotc.core.Definitions.ScalaShadowingPackage(Definitions.scala:255)
  at dotty.tools.dotc.core.Definitions.init(Definitions.scala:1496)
  at dotty.tools.dotc.core.Contexts$ContextBase.initialize(Contexts.scala:790)
  at dotty.tools.dotc.core.Contexts$Context.initialize(Contexts.scala:554)
  at dotty.tools.dotc.Run.rootContext(Run.scala:64)
  at dotty.tools.dotc.Run.<init>(Run.scala:84)
  at scala.quoted.staging.QuoteCompiler$ExprRun.<init>(QuoteCompiler.scala:102)
  at scala.quoted.staging.QuoteCompiler.newRun(QuoteCompiler.scala:45)
  at scala.quoted.staging.QuoteDriver.run(QuoteDriver.scala:40)
  at scala.quoted.staging.Toolbox$$anon$1.run(Toolbox.scala:36)
  at scala.quoted.staging.package$.run(staging.scala:19)
  at Test$.main(quote-run.scala:12)
  at Test.main(quote-run.scala)
```

(2) tests/run-staging/i6992

```
-- Error: tests/run-staging/i6992/Test_2.scala:7:16 -------------------------------
7 |    println(mcr {foo})
  |            ^^^^^^^^^
  | Exception occurred while executing macro expansion.
  | dotty.tools.dotc.MissingCoreLibraryException: Could not find package scalaShadowing from compiler core libraries.
  | Make sure the compiler core libraries are on the classpath.
```

The errors originate from a change of class loading in Java 9 (due to the module system):

https://stackoverflow.com/questions/46694600/java-9-compatability-issue-with-classloader-getsystemclassloader#comment80337394_46694600

In Java 9, the classloader is no longer a `URLClassLoader`, which breaks the following code:

https://github.com/lampepfl/dotty/blob/8eea8d8bba1da247553161e981d6d277eb375a4b/compiler/src/dotty/tools/dotc/util/ClasspathFromClassloader.scala#L10-L42

There exists a hack to achieve similar results in Java 9+

https://stackoverflow.com/questions/46519092/how-to-get-all-jars-loaded-by-a-java-application-in-java9

Related PR: #9181

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: compiler/src/dotty/tools/dotc/util/ClasspathFromClassloader.scala [hunk 1]
```diff
@@ -27,17 +27,22 @@ object ClasspathFromClassloader {
...
+              // classloaders since the current classloader might intentionally
+              // filter loading classes from its parent (for example
+              // BootFilteredLoader in the sbt launcher does this and we really
+              // don't want to include the scala-library that sbt depends on
+              // here), but we do need to look at the parent of the REPL
+              // classloader, so we special case it. We can't do this using a type
+              // test since the REPL classloader class itself is normally loaded
+              // with a different classloader.
               collectClassLoaderPaths(cl.getParent)
+            else if cl.getClass.getName.contains("AppClassLoader") then
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/util/ClasspathFromClassloader.scala:41]: can you check `cl == cl.getSystemClassLoader` instead?

Comment (smarter) [reply compiler/src/dotty/tools/dotc/util/ClasspathFromClassloader.scala:41]: or Maybe cl.getClass == cl.getSystemClassLoader.getClass, not sure.

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/util/ClasspathFromClassloader.scala:41]: Maybe `cl.getParent eq cl.getSystemClassLoader`. I will check.

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/util/ClasspathFromClassloader.scala:41]: No, the parent is the `PlatformClassLoader`. It works with `cl eq ClassLoader.getSystemClassLoader`

File: compiler/src/dotty/tools/dotc/util/ClasspathFromClassloader.scala [hunk 2]
```diff
@@ -27,17 +27,22 @@ object ClasspathFromClassloader {
...
+              // don't want to include the scala-library that sbt depends on
+              // here), but we do need to look at the parent of the REPL
+              // classloader, so we special case it. We can't do this using a type
+              // test since the REPL classloader class itself is normally loaded
+              // with a different classloader.
               collectClassLoaderPaths(cl.getParent)
+            else if cl eq ClassLoader.getSystemClassLoader then
+              // HACK: For Java 9+, if the classloader is an AppClassLoader then use the classpath from the system
+              // property `java.class.path`.
+              // See: https://java-browser.yawk.at/java/10/java.base/jdk/internal/loader/ClassLoaders.java#163
```
Comment (smarter) [inline compiler/src/dotty/tools/dotc/util/ClasspathFromClassloader.scala:44]: It's be best to not leave a link to GPL code here I think :)

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/util/ClasspathFromClassloader.scala]: Removed

**Pull request merged**